engine-types/src/types/gas.rs

```diff
@@ -87,6 +87,7 @@ impl Div<u64> for EthGas {
     type Output = Self;
 
     fn div(self, rhs: u64) -> Self::Output {
+        assert!(rhs != 0, "ZERO_IS_AN_INVALID_DENOMINATOR");
```

Comment:
Instead of panicking, we could enforce this at the type level by using `rhs: NonZeroU64`. This requires changes in other places as well to map into the non-zero type, but it's not that bad:

```rust
// engine-precompiles/src/alt_bn256.rs

let pair_element_len = NonZeroUsize::try_from(consts::PAIR_ELEMENT_LEN)
    .and_then(NonZeroU64::try_from)
    .map_err(utils::err_usize_conv)?;
```

Changing the type would technically be a public API change, but again I don't think that is a big deal because I doubt anyone is using the library and if they are then changing the call-site is not that hard.

Reply:
Also, I know were are avoiding unsafe conversions (which is why the code in `https://crates.io/crates/const_guards` uses `try_from` conversions), but it looks a little silly when it is a compile-time known constant we are converting.

Reply:
You mean replace `impl Div<u64> for EthGas` with `impl Div<NonZeroU64> for EthGas` ?
cc @0x3bfc